src/doom/m_menu.c

```diff
@@ -634,8 +635,19 @@ void M_DoSave(int slot)
 //
 static void SetDefaultSaveName(int slot)
 {
-    M_snprintf(savegamestrings[itemOn], SAVESTRINGSIZE - 1,
-               "JOYSTICK SLOT %i", itemOn + 1);
+    // map from IWAD or PWAD?
+    if (maplumpinfo->wad_file == lumpinfo[0]->wad_file)
+    {
+        M_snprintf(savegamestrings[itemOn], SAVESTRINGSIZE - 1,
```

Comment:
This should actually be `SAVESTRINGSIZE` (no `- 1`) - `M_snprintf()` takes the size of the buffer, not the string length.